### PR TITLE
Install add-ons in user's local site-packages

### PR DESCRIPTION
**Pull request by kernc**

### CODE REVIEW

**Comment by lanzagar** [comment]
Orange still check if I have root privileges, shows an error and does not let me manage addons.
See [line 418](https://github.com/biolab/orange3/pull/971/files#diff-55a6a7f617395a75195203c03b442985R418).

**Comment by kernc** [comment]
Gah, I meant to delete that! Thanks. How about now?

**Comment by lanzagar** [comment]
Works for me, but not for people running it in virtualenv:

pip.exceptions.InstallationError: Can not perform a '--user' install. User site-packages are not visible in this virtualenv.

I wanted to suggest this anyway and it might solve the problem as well:
Should we check if we have permission to install addons as previously (works in virtualenv and if run as root). That way an addon can be installed system-wide (useful if lots of users want to use it on a shared computer).
And if not, only then switch to the `--user` installation.

File: Orange/canvas/application/addons.py [hunk 1]
```diff
@@ -565,6 +557,7 @@ def __on_installer_error(self, command, pkg, retcode, output):
 
     def __on_installer_finished(self):
         message_information(
+            "Changes successfully applied in <i>{}</i>.<br>".format(USER_SITE) +
```
Comment (lanzagar) [inline Orange/canvas/application/addons.py]: This now prints that the changes have been made in the local user dir even if `user_install` is False and they were not.

**Comment by lanzagar** [comment]
Although this pull request is probably not the most appropriate one to add Orange3-Associate the bigger question is - is this addon now finished and production ready enough that we want to have it explicitly listed?
All addons on pypi show up in the list anyway, right?

File: Orange/canvas/application/addons.py (hunk 1 above)
Comment (kernc) [inline Orange/canvas/application/addons.py]: Thanks, please see again.

**Comment by kernc** [comment]
It's my opinion that Orange3-Associate is as good as ever. Anyone is welcome to try it out. :stuck_out_tongue_winking_eye:

**Pull request merged**